io/shared/src/test/scala/fs2/io/net/tcp/SocketSuite.scala

```diff
@@ -216,5 +217,47 @@ class SocketSuite extends Fs2Suite with SocketSuitePlatform {
           assertEquals(msgs, Vector(msg))
         }
     }
+
+    test("read after timed out read not allowed on JVM") {
+      val setup = for {
+        serverSetup <- Network[IO].serverResource(Some(ip"127.0.0.1"))
+        (bindAddress, server) = serverSetup
+        client <- Network[IO].client(bindAddress)
+      } yield (server, client)
+      Stream
+        .resource(setup)
+        .flatMap { case (server, client) =>
+          val echoServer = server.flatMap(c => c.reads.through(c.writes))
+          val msg = Chunk.array("Hello!".getBytes)
+          val prg =
+            client.write(msg) *>
+              client.readN(msg.size) *>
+              client.readN(msg.size).timeout(100.millis).recover { case _: TimeoutException =>
+                Chunk.empty
+              } *>
+              client.write(msg) *>
+              client
+                .readN(msg.size)
+                .timeout(1000.millis)
```

Comment:
Why is another timeout needed here? For Node.js?

Reply:
Oh that can come out now. That was originally needed when we were hanging the 2nd+ request.

Reply:
No, Node.js succeeds below.

Reply:
```suggestion
```